### DISCUSSION TITLE
How to get all the projects in harbor

### DISCUSSION THREAD

**Discussion by finalha** [goharbor/harbor#17426] [Q&A] [answered]
The version of harbor is  v2.2.4-1bc848b3 and I want to get all the projects in the harbor.
Try to use the shell command below:
`curl -u $Harbor_User:$Harbor_Passwd  -H "Content-Type: application/json" -X GET  http://${Harbor_Address}/api/v2.0/projects  -k  | python -m json.tool | grep name | awk '/"name": /' | awk -F '"' '{print $4}'`
But only return few projects. How to get all the projects in harbor with shell?

**Comment by dioguerra** [comment, marked as answer]
```yaml
PAGE=1
      while : ; do
        # Get page # of all registry projects
        PROJECTS=$(curl -k -X 'GET' "{{ .Values.harbor.externalURL }}/api/v2.0/projects?page=${PAGE}&page_size=25&with_detail=true" -H 'authorization: Basic {{ printf "admin:%s" .Values.harbor.harborAdminPassword | b64enc }}' -H 'accept: application/json')
        `((PAGE+=1))`

        # Exit if no more projects
        if [ $(echo $PROJECTS | jq length) -eq 0 ]; then
          wait
          break
        fi

        [CODE HERE]
done
```

**Reply by finalha** [reply]
Thanks, it works. But it seems only to get public projects, how to get private projects?

**Reply by dioguerra** [reply]
your account needs to have access to the private projects.

**Reply by IMpcuong** [reply]
> `{{ .Values.harbor.externalURL }}`

I have used this syntax for my script, but unfortunately, Bash has complained about the double curly braces. My question is: in which circumstance (language, etc) can I utilize your code snippet instead of hardcoding my private registry? TYSM.

**Reply by dioguerra** [reply]
This is Helm template form using golang template language. Just replace the {{ .VALUES }} with a fixed value or $VARIABLE
